### PR TITLE
PUBDEV-6091 - expose timestamp in Python & R API

### PR DESCRIPTION
**Pull request by Pscheidl**
JIRA: https://0xdata.atlassian.net/browse/PUBDEV-6091

### CODE REVIEW

**Comment by Pscheidl** [comment]
Note to self: To be discussed with @grigory93 first !

File: h2o-r/tests/testdir_jira/runit_pubdev_6019.R [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+test.timestamp <- function() {
+  prostate.data <- h2o.importFile(path = locate('smalldata/logreg/prostate_train.csv'))
+  gbm.model <- h2o.gbm(x=c("AGE","RACE","DPROS"),y="CAPSULE",training_frame=prostate.data ,model_id="gbm_prostate_model", ntrees = 1, max_depth = 1, seed = 1)
+  expect_false(is.null(gbm.model@model$timestamp))
+
+}
+
+doTest("H2OTree visitor", test.timestamp)
```
Comment (ledell) [inline h2o-r/tests/testdir_jira/runit_pubdev_6019.R]:
```suggestion
doTest("Test model timestamp", test.timestamp)
```

**Comment by ledell** [comment]
@Pscheidl So.... I think the model already has model start time information stored in `start_time`.  In R it does, at least:

```
> gbm.model@model$timestamp
$timestamp
[1] 1.544048e+12
> gbm.model@model$start_time
[1] 1.544048e+12
> gbm.model@model$end_time
[1] 1.544048e+12
> gbm.model@model$run_time
[1] 146
```

**Comment by Pscheidl** [comment]
Sorry, this PR is a mess I admit. I was doing it in a rush in Amsterdam and that's not good.

1. I deleted R API changes.
2. The exposure in python is still needed, so I made it reflect R API at least.
3. Grigory wants it in the summary table. Is everyone ok with it ? @ledell @michalkurka

**Comment by michalkurka** [comment]
@grigory93 can you please explain why is a timestamp helpful in the grid summary? I don't fully understand the use case of multiple runs.

If there are worries about making it too cluttered we can add a parameter "verbose" to the grid summary method or something along those lines.

General guidance is to optimize for the typical use case. Lets see what Gregory has to say about the multiple runs but if it is an edge case then I would agree with @ledell .

**Comment by Pscheidl** [comment]
@michalk see what gregory said ^^^^

Copy&paste:
```plain
When getting models added to the grid object on multiple runs and analyzing its summary it's not clear when the models were ran (created), that's why timestamp would have been helpful to me (and I expect to others who are re-using grid objects).
```

**Comment by ledell** [comment]
I also don't see a huge value in returning timestamp by default - I'd prefer a tidy grid summary object with just the most essential info: model id and param values.

**Comment by ledell** [comment]
@Pscheidl If Python models don't have the `start_time` `end_time` and `run_time` objects, then it would be good to add them (I haven't checked if they are already there).

**Comment by ledell** [comment]
@Pscheidl Ok, sounds good -- let's add these to Python (only) then.

File: h2o-py/h2o/estimators/estimator_base.py [hunk 2]
```diff
@@ -257,6 +257,9 @@ def _resolve_model(self, model_id, model_json):
         m._metrics_class = metrics_class
         m._parms = self._parms
         m._estimator_type = self._estimator_type
+        m._start_time = model_json.get('output', {}).get('start_time', None)
```
Comment (michalkurka) [inline h2o-py/h2o/estimators/estimator_base.py]: just to make are these always present?

Comment (Pscheidl) [reply h2o-py/h2o/estimators/estimator_base.py]: If not, the outcome is None. But yes, these are added to all models.

Comment (Pscheidl) [reply h2o-py/h2o/estimators/estimator_base.py]: Model has methods startClock and stopClock. As `Model` is parent to all other mdoels, these are always in place.

**Pull request merged**